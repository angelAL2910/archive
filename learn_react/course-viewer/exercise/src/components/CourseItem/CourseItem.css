.delete{
    border: 2px solid red;
    color: red;
    background-color: white;
}

.watch-button{
    background-color: #e8e8e8;
    border: 0;
    border-radius: 5px;
}