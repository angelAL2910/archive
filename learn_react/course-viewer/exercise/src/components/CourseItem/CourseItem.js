import React from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { deleteCourseAction } from "../../actions/actionCreators";
import "./CourseItem.css";

export class CourseItem extends React.Component {
  editThisCourse(e) {
    this.props.addCourseToStoreForEditing(this.props.course);
  }
  deleteThisCourse(e) {
    this.props.deleteCourse(this.props.course.id);
  }
  render() {
    let { course, authors } = this.props;
    let authorName = "";
    let path = "/courses/" + course.slug;
    authors.forEach(author => {
      if (author.id == course.authorId) {
        authorName = author.name;
      }
    });
    return (
      <tr>
        <td>
          <button className="watch-button">Watch</button>
        </td>
        <td onClick={this.editThisCourse.bind(this)} className="title">
          <Link to={path}>{course.title}</Link>
        </td>
        <td>{authorName}</td>
        <td>{course.category}</td>
        <td>
          <button onClick={this.deleteThisCourse.bind(this)} className="delete">
            <Link to="/courses">Delete</Link>
          </button>
        </td>
      </tr>
    );
  }
}

export const mapDispatchToPtops = dispatch => ({
  deleteCourse: id => {
    dispatch(deleteCourseAction(id));
  }
});

export default connect(
  null,
  mapDispatchToPtops
)(CourseItem);
