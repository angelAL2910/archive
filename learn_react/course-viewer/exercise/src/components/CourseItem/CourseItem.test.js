import React from "react";
import renderer from "react-test-renderer";
import { mapDispatchToPtops, CourseItem } from "./CourseItem";
import { BrowserRouter } from "react-router-dom";

const fakeStore = {
  courses: [
    {
      id: 1,
      title: "Securing React Apps with Auth01",
      slug: "Securing-React-Apps-with-Auth01",
      authorId: "2",
      category: "JavaScript"
    },
    {
        id: 2,
        title: "Securing React Apps with Auth02",
        slug: "Securing-React-Apps-with-Auth02",
        authorId: "2",
        category: "JavaScript2"
      },
      {
        id: 3,
        title: "Securing React Apps with Auth03",
        slug: "Securing-React-Apps-with-Auth03",
        authorId: "2",
        category: "JavaScript3"
      },
      {
        id: 4,
        title: "Securing React Apps with Auth04",
        slug: "Securing-React-Apps-with-Auth04",
        authorId: "2",
        category: "JavaScript4"
      }
  ],
  authors: [
    {
      id: 1,
      name: "Cory House"
    }
  ],
  editCourse: [
    {
      id: 1,
      title: "Securing React Apps with Auth01",
      slug: "Securing-React-Apps-with-Auth01",
      authorId: "2",
      category: "JavaScript"
    }
  ],
  routing: {
    locationBeforeTransitions: null
  },
};

test("Testing the courseitem", () => {
  let dispatchedActions = [];

  let newCourse = {
    id: 5,
    title: "Securing React Apps with Auth012",
    slug: "Securing-React-Apps-with-Auth012",
    authorId: "2",
    category: "JavaScript2"
  };
  //expect(mapStateToProps(fakeStore).authors.length).toEqual(1);
  const dispatch = jest.fn();
  mapDispatchToPtops(dispatch).deleteCourse(1);
  expect(dispatch.mock.calls[0][0]).toEqual({ type: 'DELETE_COURSE', id: 1});
//   expect(dispatchedActions.length).toEqual(1);
  //expect(fakeStore.courses).toContain(newCourse);
});

test("Protecting against regression issues.....", () => {
  let tree = renderer.create(
      <BrowserRouter>
        <CourseItem course={fakeStore.courses[0]} authors={fakeStore.authors} />
      </BrowserRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();

});
