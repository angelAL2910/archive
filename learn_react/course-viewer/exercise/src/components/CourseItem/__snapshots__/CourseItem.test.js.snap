// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Protecting against regression issues..... 1`] = `
<tr>
  <td>
    <button
      className="watch-button"
    >
      Watch
    </button>
  </td>
  <td
    className="title"
    onClick={[Function]}
  >
    <a
      href="/courses/Securing-React-Apps-with-Auth01"
      onClick={[Function]}
    >
      Securing React Apps with Auth01
    </a>
  </td>
  <td>
    
  </td>
  <td>
    JavaScript
  </td>
  <td>
    <button
      className="delete"
      onClick={[Function]}
    >
      <a
        href="/courses"
        onClick={[Function]}
      >
        Delete
      </a>
    </button>
  </td>
</tr>
`;
